main.cpp

```diff
@@ -259,72 +250,104 @@ struct HeapA
     }
 };
 
-#include <iostream>
 struct IntType;
 struct DoubleType;
 
 struct FloatType
 {
 
-    FloatType(float num) : value( new float (num) ){}
+    private:
+        float* value;
+        FloatType& powInternal(float arg);
 
-    ~FloatType()
-    {
-        delete value;
-        value = nullptr;
-    }
+    public:    
```

Comment:
if you move these private members to the end of the class, you can eliminate this `public` keyword, since struct is public by default
less typing = faster coding
- [ ] revise